pkg/cmd/deployment/create/create.go

```diff
@@ -0,0 +1,191 @@
+package create
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmdutil"
+	"github.com/cli/cli/v2/pkg/iostreams"
+	"github.com/spf13/cobra"
+)
+
+type createOptions struct {
+	BaseRepo   func() (ghrepo.Interface, error)
+	Branch     func() (string, error)
+	HttpClient func() (*http.Client, error)
+	IO         *iostreams.IOStreams
+
+	Ref         string
+	Environment string
+	Status      string
+}
+
+type deployment struct {
+	ID int `json:"id"`
+}
+
+func NewCmdCreate(f *cmdutil.Factory, runF func(*createOptions) error) *cobra.Command {
+	opts := createOptions{
+		HttpClient: f.HttpClient,
+		IO:         f.IOStreams,
+		Branch:     f.Branch,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a new deployment",
+		Long: heredoc.Docf(`
+			Create a new deployment.
+
+			This will dispatch a %[1]sdeployment%[1]s event that external services can listen for and act on.
```

Comment:
**Question**

Should what creating a deployment does be documented here? Seems like maybe the concern of the github docs?

Reply:
Happy to keep this minimal. This line was stolen from https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28#about-deployments. Would "Create a new deployment." be enough?

Thinking back, I included this to try and help clarify that this is related to the GitHub Deployment API, given how "deployment" is such a generic term otherwise.